python/ray/data/tests/test_iterator.py

```diff
@@ -196,6 +208,183 @@ def collate_fn(batch: Dict[str, np.ndarray]):
+
+        def __call__(self, batch: pyarrow.Table) -> torch.Tensor:
+            """Add 5 to the "id" column at the Arrow level."""
+            modified_batch = pyarrow.Table.from_arrays(
+                [pyarrow.compute.add(batch["id"], 5)], names=["id"]
+            )
+
+            return arrow_batch_to_tensors(
+                modified_batch,
+                dtypes=self.dtypes,
+                combine_chunks=self.device == "cpu",
+            )["id"]
+
+    class TupleArrowBatchCollateFn(ArrowBatchCollateFn):
+        def __init__(
+            self,
+            dtypes: Optional[Dict[str, torch.dtype]] = None,
+            device: Optional[str] = None,
+        ):
+            self.dtypes = dtypes
+            self.device = device
+
+        def __call__(self, batch: pyarrow.Table) -> Tuple[torch.Tensor, torch.Tensor]:
+            """Add 5 to the "id" column and return both id and value."""
+            modified_batch = pyarrow.Table.from_arrays(
+                [pyarrow.compute.add(batch["id"], 5), batch["id"]],
+                names=["id", "value"],
+            )
+            tensors = arrow_batch_to_tensors(
+                modified_batch,
+                dtypes=self.dtypes,
+                combine_chunks=self.device == "cpu",
+            )
+            return tensors["id"], tensors["value"]
+
+    class DictArrowBatchCollateFn(ArrowBatchCollateFn):
+        def __init__(
+            self,
+            dtypes: Optional[Dict[str, torch.dtype]] = None,
+            device: Optional[str] = None,
+        ):
+            self.dtypes = dtypes
+            self.device = device
+
+        def __call__(self, batch: pyarrow.Table) -> Dict[str, torch.Tensor]:
+            """Add 5 to the "id" column and return both columns as dict."""
+            modified_batch = pyarrow.Table.from_arrays(
+                [pyarrow.compute.add(batch["id"], 5), batch["id"]],
+                names=["id", "value"],
+            )
+            return arrow_batch_to_tensors(
+                modified_batch,
+                dtypes=self.dtypes,
+                combine_chunks=self.device == "cpu",
+            )
+
+    class ListArrowBatchCollateFn(ArrowBatchCollateFn):
+        def __init__(
+            self,
+            dtypes: Optional[Dict[str, torch.dtype]] = None,
+            device: Optional[str] = None,
+        ):
+            self.dtypes = dtypes
+            self.device = device
+
+        def __call__(self, batch: pyarrow.Table) -> List[torch.Tensor]:
+            """Add 5 to the "id" column and return both columns as list."""
+            modified_batch = pyarrow.Table.from_arrays(
+                [pyarrow.compute.add(batch["id"], 5), batch["id"]],
+                names=["id", "value"],
+            )
+            tensors = arrow_batch_to_tensors(
+                modified_batch,
+                dtypes=self.dtypes,
+                combine_chunks=self.device == "cpu",
+            )
+            return [tensors["id"], tensors["value"]]
```

Comment:
Looks like there is a lot of code duplication among these classes. 
Let's consolidate them.
Also, dtypes is always None and device is always "cpu". We can just remove the init args.